python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -472,3 +482,153 @@ def oldScarletToData(blend: Blend, psfCenter: tuple[int, int], xy0: Point2I):
     )
 
     return blendData
+
+
+class ScarletModelFormatter(TypelessFormatter):
+    """Read and write zip archives.
+
+    In order for files to be read from a zip file, the pydantic model
+    must have a `load` method that accepts a file object and the filename
+    as arguments, as well as a `from_zip` method that accepts a dictionary
+    of files to create the object from the zip archive.
+    """
+
+    default_extension = ".scarlet"
+    unsupported_parameters = None
+    can_read_from_uri = True
+
+    def _extract_all_files(self, uri: ResourcePath) -> dict[str, Any]:
+        """
+        Extract all files from a ZIP archive.
+
+        Parameters
+        ----------
+        uri : `ResourcePath`
+            URI of the ZIP file
+
+        Returns
+        -------
+        files :
+            Dictionary mapping filenames to the desired file type.
+        """
+        result = {}
+        with uri.open("rb") as f:
+            with zipfile.ZipFile(f, 'r') as zipf:
+                for filename in zipf.namelist():
+                    with zipf.open(filename) as f:
+                        result[filename] = from_json(f.read())
+        return result
+
+    # Extracting a single file from the ZIP
+    def _extract_blend(self, uri: ResourcePath, blend_filename: str) -> dict[str, Any]:
```

Comment:
Consider making this the blend_id or blend_ids rather than the explicit filename. The code in here can understand how to go from blend to zip member.